src/models/classes/factoryorder.js

```diff
@@ -3,8 +3,8 @@ class FactoryOrder extends OrderCore
     constructor(products)
     {
         var time = GAME.model.config.orderTransportDuration;
-        var id = GAME.model.orders.length;
+        super(products, time);
 
-        super(products, time, id);
+        this.id = GAME.model.orders.length;
```

Comment:
Without the super call, things like `this.orderCost` etc. won't be initialised. Hence that'd be a bug.

So the super call is required.